code/ai/aicode.cpp

```diff
@@ -13972,36 +13970,37 @@ void ai_frame(int objnum)
 	Pl_objp->phys_info.flags &= ~PF_GLIDING;
 
 	// warping out?
-	if ((aip->mode == AIM_WARP_OUT) || (aip->ai_flags & AIF_TRYING_UNSUCCESSFULLY_TO_WARP))
+	if ((aip->mode == AIM_WARP_OUT) || (aip->ai_flags[AI::AI_Flags::Trying_unsuccessfully_to_warp]))
 	{
 		ai_warp_out(Pl_objp);
 
 		// Goober5000 - either we were never trying unsuccessfully, or we were but now
 		// we're successful... in either case, since we're actually warping we simply return
-		if (!(aip->ai_flags & AIF_TRYING_UNSUCCESSFULLY_TO_WARP))
+		if (!(aip->ai_flags[AI::AI_Flags::Trying_unsuccessfully_to_warp]))
 			return;
 	}
 
 	ai_maybe_self_destruct(Pl_objp, aip);
 	ai_process_mission_orders( objnum, aip );
 
 	//	Avoid a shockwave, if necessary.  If a shockwave and rearming, stop rearming.
-	if (aip->ai_flags & AIF_AVOID_SHOCKWAVE) {
+	if (aip->ai_flags[AI::AI_Flags::Avoid_shockwave_ship, AI::AI_Flags::Avoid_shockwave_weapon]) {
 		if (ai_avoid_shockwave(Pl_objp, aip)) {
-			aip->ai_flags &= ~(AIF_BIG_SHIP_COLLIDE_RECOVER_1 | AIF_BIG_SHIP_COLLIDE_RECOVER_2);
-			if (aip->ai_flags & (AIF_AWAITING_REPAIR | AIF_BEING_REPAIRED))
+			aip->ai_flags.remove(AI::AI_Flags::Big_ship_collide_recover_1);
+			aip->ai_flags.set(AI::AI_Flags::Big_ship_collide_recover_2);
```

Comment:
The original code removes both flags but the new code only removes one and sets one.